master/internal/api_generic_tasks.go

```diff
@@ -86,7 +91,13 @@ func (a *apiServer) getGenericTaskLaunchParameters(
 	taskConfig := model.DefaultConfigGenericTaskConfig(&taskSpec.TaskContainerDefaults)
 	workDirInDefaults := taskConfig.WorkDir
 
-	if err := yaml.Unmarshal([]byte(configYAML), &taskConfig); err != nil {
+	if len(parentConfig) != 0 && len(configYAML) != 0 {
+		if err := yaml.Unmarshal(parentConfig, &taskConfig); err != nil {
+			return nil, nil, nil, fmt.Errorf("yaml unmarshaling generic task config: %w", err)
+		}
+	}
+
+	if err := yaml.Unmarshal(configBytes, &taskConfig); err != nil {
```

Comment:
something seems off here; at this point configBytes will either be the full parentConfig or the full configBytes, depending on what was passed, but if it was the parent config we're just marshaling it a second time. what is this code supposed to do exactly?

Reply:
ConfigBytes will only be filled with the parent config when `len(configYAML) == 0`, because of this the parentConfig unmarshalling in the above lines will be skipped.

Reply:
but we do the unmarshaling twice when parentConfig != 0 right? and we do it twice with the same bytes.

Reply:
the only case when we unmarshal twice is when both parentConfig and configYAML are filled. This occurs in the case where we are forking from another task but we are also including some changes in the config. These changes would exist in configYAML. We can talk offline if you want to discuss this functionality.

Reply:
Yeah let's talk. I still don't still the point of unmarshaling the same thing twice in that case.

Reply:
duh ok i see now. but yeah let's chat offline. wow this really.. uh.. confused me to read.

Reply:
sorry, got confused for a second, i read this one too many times and everything started to run together.

when parentConfig and configBytes are both not nil, we marshal parentConfig in once and then configYAML on top of it. yeah, i wasn't expecting merge behavior but maybe that's fine.